### ISSUE TITLE
Pod name in artifacts/k8s.log.txt is not consistent with buid-log.txt and k8s.dump.txt

### ISSUE THREAD

**Issue by nak3** [knative/test-infra#1516]
TL;DR: In short, k8s.log.txt does not have correct pod logs in test grid.

k8s.dump.txt has `scale-to-n-scale-10-0-fwxgxqlo` in the log.
build-log.txt also has `scale-to-n-scale-10-0-fwxgxqlo` in the log.
__BUT__
k8s.log.txt has `scale-to-n-scale-10-0-afboppjo` in the log.

---
For example, download artifacts from https://prow.knative.dev/view/gcs/knative-prow/logs/ci-knative-serving-istio-1.2-no-mesh/1190871668113805312 :

https://storage.googleapis.com/knative-prow/logs/ci-knative-serving-istio-1.2-no-mesh/1190871668113805312/artifacts/k8s.dump.txt
https://storage.googleapis.com/knative-prow/logs/ci-knative-serving-istio-1.2-no-mesh/1190871668113805312/build-log.txt
https://storage.googleapis.com/knative-prow/logs/ci-knative-serving-istio-1.2-no-mesh/1190871668113805312/artifacts/k8s.log.txt

k8s.dump.txt has the event logs:

```
pod/scale-to-n-scale-10-0-fwxgxqlo-n2t8r-deployment-845c98456-m556z                     Started container user-container
```

So, ksvc `scale-to-n-scale-10-0-fwxgxqlo` must be created.

As expected, build-log.txt has it.

```
        scale.go:166: scale-to-n-scale-10-0-fwxgxqlo is ready.
```

But k8s.log.txt shows different pod's hash `scale-to-n-scale-10-0-afboppjo-gztx9`.

```
serving-tests/scale-to-n-scale-10-0-afboppjo-gztx9-deployment-dd7cdb48d-44qhx[user-container]: 2019/11/03 07:02:51 Hello world received a request.
```

__NOTE__

- System pods like activator, autoscaler etc has correct pod name.
- If checking PR's test result, it looks fine. But only [test grid](https://testgrid.knative.dev/serving#istio-1.2-no-mesh)'s k8s.log.txt seems wrong.

**Comment by adrcunha** [comment]
TestGrid points to the same artifacts that Spyglass shows, so it's not TestGrid's fault.

k8s.log.txt and k8s.dump.txt are generated differently, and k8s.log.txt is specific to Knative serving:

* k8s.dump.txt is generated by https://github.com/knative/test-infra/blob/e7ee6c7ed4f4ff1c24902e714d874310db23cf36/scripts/e2e-tests.sh#L97
* k8s.log.txt is generated by https://github.com/knative/serving/blob/fee536a1ea1e91386c622727fd08486216e62e9b/test/e2e-common.sh#L334

kail was added by https://github.com/knative/serving/pull/4843 and slightly modified in https://github.com/knative/serving/pull/4906

@JRBANCEL or @tcnghia might have a clue why the differences in pod names.

I'll move this to knative/serving since it's a more appropriate location.
